hutch_python/utils.py

```diff
@@ -433,3 +436,75 @@ def get_fully_qualified_domain_name():
         )
         logger.debug("getfqdn failed", exc_info=True)
         return ""
+
+
+abort_msg = """
+Your RunEngine has been interrupted.  If you aborted the current run (Ctrl+C),
+the RunEngine is ready for use.  If you paused the run (Ctrl+\\), you must
+resume, abort, stop, halt the RunEngine before continuing.
+"""
+
+
+class AbortSigintHandler(SignalHandler):
+    """
+    Interprets SIGINT (Ctrl+C) as an abort request.
+    """
+    def __init__(self, RE):
+        super().__init__(signal.SIGINT, log=RE.log)
+        self.RE = RE
+
+    def handle_signals(self):
+        # Check for pause requests from keyboard.
+        if self.RE.state.is_running:
+            self.RE.stop()
+            # Use our own logging
+            logger.warning("Aborting current run.")
+
+
+class SigquitHandler(SignalHandler):
+    """
+    Interprets SIGQUIT (Ctrl+\\) as standard RunEngine interrupt behavior.
+    Copied verbatim from bluesky.utils.SigintHandler, lacking a way to
+    subclassing the original while changing the signal type.
+    """
+    def __init__(self, RE):
+        super().__init__(signal.SIGQUIT, log=RE.log)
+        self.RE = RE
+        self.last_sigint_time = None  # time most recent SIGQUIT was processed
+        self.num_sigints_processed = 0  # count SIGQUITs processed
+
+    def __enter__(self):
+        return super().__enter__()
+
+    def handle_signals(self):
+        # Check for pause requests from keyboard.
+        # TODO, there is a possible race condition between the two
```

Comment:
So the race condition is why we check the time to ensure we don't see two `c-\` for one keypress?

Reply:
That's my understanding, yes.